streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3696,17 +3656,15 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
   /**
    * Creates a stream from an effect producing a value of type `A` which repeats using the specified schedule
    */
-  def repeatEffectWith[R, E, A](effect: ZIO[R, E, A], schedule: Schedule[R, A, _]): ZStream[R, E, A] =
-    fromEffect(schedule.initial).flatMap { initial =>
-      unfoldM(initial) { state =>
-        effect.flatMap(value => schedule.update(value, state).fold(_ => None, state => Some((value, state))))
-      }
+  def repeatEffectWith[R, E, A](effect: ZIO[R, E, A], schedule: Schedule[R, A, Any]): ZStream[R with Clock, E, A] =
+    ZStream {
+      schedule.driver.toManaged_.map(driver => effect.mapError(Some(_)).tap(driver.next).map(Chunk(_)))
```

Comment:
This was updated in master. The current implementation is:
```scala
    ZStream.fromEffect(schedule.initial zip effect).flatMap {
      case (initialScheduleState, value) =>
        ZStream.succeed(value) ++ ZStream.unfoldM((initialScheduleState, value)) {
          case (state, lastValue) =>
            schedule
              .update(lastValue, state)
              .foldM(_ => ZIO.succeed(Option.empty), newState => effect.map(value => Some((value, (newState, value)))))
        }
    }
```

The main difference is that it executes the effect once without any delays, and then starts stepping the schedule. I think translated to the driver formulation that would be (untested):
```scala
ZStream.fromEffect(effect zip schedule.driver).flatMap { case (a, driver) =>
  ZStream(a) ++ 
  ZStream.repeatEffect(effect.mapError(Some(_)).tap(driver.next))
}
```

Reply:
Fixed